src/category/service.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  BadRequestException,
+  Injectable,
+  InternalServerErrorException,
+} from '@nestjs/common';
+import { Prisma } from '@prisma/client';
+import { PrismaService } from 'src/prisma/service';
+
+@Injectable()
+export class CategoryService {
+  constructor(private readonly prisma: PrismaService) {}
+  async createCategory(dto: Prisma.CategoryCreateInput) {
+    try {
+      const category = await this.prisma.category.create({ data: dto });
+      return category;
+    } catch (error) {
+      throw new BadRequestException(error);
+    }
+  }
+  async updateCategory(id: number, dto: Prisma.CategoryUpdateInput) {
+    try {
+      const category = await this.prisma.category.update({
+        where: {
+          id: +id,
```

Comment:
No need, id should be passed as a number into this function, parsing can be done in the controller or sth else.